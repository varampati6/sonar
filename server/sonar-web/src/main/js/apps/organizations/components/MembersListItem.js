/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
//@flow
import React from 'react';
import Avatar from '../../../components/ui/Avatar';
import { translateWithParameters } from '../../../helpers/l10n';
import { formatMeasure } from '../../../helpers/measures';
import RemoveMemberForm from './forms/RemoveMemberForm';
import ManageMemberGroupsForm from './forms/ManageMemberGroupsForm';
import type { Member } from '../../../store/organizationsMembers/actions';
import type { Organization, OrgGroup } from '../../../store/organizations/duck';

type Props = {
  member: Member,
  organization: Organization,
  organizationGroups: Array<OrgGroup>,
  removeMember: Member => void,
  updateMemberGroups: (member: Member, add: Array<string>, remove: Array<string>) => void
};

const AVATAR_SIZE: number = 36;

export default class MembersListItem extends React.PureComponent {
  props: Props;

  render() {
    const { member, organization } = this.props;
    return (
      <tr>
        <td className="thin nowrap">
          <Avatar hash={member.avatar} email={member.email} name={member.name} size={AVATAR_SIZE} />
        </td>
        <td className="nowrap text-middle">
          <strong>{member.name}</strong>
          <span className="note little-spacer-left">{member.login}</span>
        </td>
        {organization.canAdmin &&
          <td className="text-right text-middle">
            {translateWithParameters(
              'organization.members.x_groups',
              formatMeasure(member.groupCount || 0, 'INT')
            )}
          </td>}
        {organization.canAdmin &&
          <td className="nowrap text-middle text-right">
            <div className="dropdown">
              <button className="dropdown-toggle little-spacer-right" data-toggle="dropdown">
                <i className="icon-settings" />
                {' '}
                <i className="icon-dropdown" />
              </button>
              <ul className="dropdown-menu dropdown-menu-right">
                <li>
                  <ManageMemberGroupsForm
                    organizationGroups={this.props.organizationGroups}
                    organization={this.props.organization}
                    updateMemberGroups={this.props.updateMemberGroups}
                    member={this.props.member}
                  />
                </li>
                <li role="separator" className="divider" />
                <li>
                  <RemoveMemberForm
                    organization={this.props.organization}
                    removeMember={this.props.removeMember}
                    member={this.props.member}
                  />
                </li>
              </ul>
            </div>
          </td>}
      </tr>
    );
  }
}
