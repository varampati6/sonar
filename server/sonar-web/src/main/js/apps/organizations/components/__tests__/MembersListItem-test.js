/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import React from 'react';
import { shallow } from 'enzyme';
import MembersListItem from '../MembersListItem';

const organization = { key: 'foo', name: 'Foo' };
const admin = { login: 'admin', name: 'Admin Istrator', avatar: '', groupCount: 3 };
const john = { login: 'john', name: 'John Doe', avatar: '7daf6c79d4802916d83f6266e24850af' };

it('should not render actions and groups for non admin', () => {
  const wrapper = shallow(<MembersListItem organization={organization} member={admin} />);
  expect(wrapper).toMatchSnapshot();
});

it('should render actions and groups for admin', () => {
  const wrapper = shallow(
    <MembersListItem organization={{ ...organization, canAdmin: true }} member={admin} />
  );
  expect(wrapper).toMatchSnapshot();
});

it('should groups at 0 if the groupCount field is not defined (just added user)', () => {
  const wrapper = shallow(
    <MembersListItem organization={{ ...organization, canAdmin: true }} member={john} />
  );
  expect(wrapper).toMatchSnapshot();
});
