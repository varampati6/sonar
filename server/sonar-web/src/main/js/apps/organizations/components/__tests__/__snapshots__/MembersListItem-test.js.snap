// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should groups at 0 if the groupCount field is not defined (just added user) 1`] = `
<tr>
  <td
    className="thin nowrap"
  >
    <Connect(Avatar)
      hash="7daf6c79d4802916d83f6266e24850af"
      name="John Doe"
      size={36}
    />
  </td>
  <td
    className="nowrap text-middle"
  >
    <strong>
      John Doe
    </strong>
    <span
      className="note little-spacer-left"
    >
      john
    </span>
  </td>
  <td
    className="text-right text-middle"
  >
    organization.members.x_groups.0
  </td>
  <td
    className="nowrap text-middle text-right"
  >
    <div
      className="dropdown"
    >
      <button
        className="dropdown-toggle little-spacer-right"
        data-toggle="dropdown"
      >
        <i
          className="icon-settings"
        />
         
        <i
          className="icon-dropdown"
        />
      </button>
      <ul
        className="dropdown-menu dropdown-menu-right"
      >
        <li>
          <ManageMemberGroupsForm
            member={
              Object {
                "avatar": "7daf6c79d4802916d83f6266e24850af",
                "login": "john",
                "name": "John Doe",
              }
            }
            organization={
              Object {
                "canAdmin": true,
                "key": "foo",
                "name": "Foo",
              }
            }
          />
        </li>
        <li
          className="divider"
          role="separator"
        />
        <li>
          <RemoveMemberForm
            member={
              Object {
                "avatar": "7daf6c79d4802916d83f6266e24850af",
                "login": "john",
                "name": "John Doe",
              }
            }
            organization={
              Object {
                "canAdmin": true,
                "key": "foo",
                "name": "Foo",
              }
            }
          />
        </li>
      </ul>
    </div>
  </td>
</tr>
`;

exports[`should not render actions and groups for non admin 1`] = `
<tr>
  <td
    className="thin nowrap"
  >
    <Connect(Avatar)
      hash=""
      name="Admin Istrator"
      size={36}
    />
  </td>
  <td
    className="nowrap text-middle"
  >
    <strong>
      Admin Istrator
    </strong>
    <span
      className="note little-spacer-left"
    >
      admin
    </span>
  </td>
</tr>
`;

exports[`should render actions and groups for admin 1`] = `
<tr>
  <td
    className="thin nowrap"
  >
    <Connect(Avatar)
      hash=""
      name="Admin Istrator"
      size={36}
    />
  </td>
  <td
    className="nowrap text-middle"
  >
    <strong>
      Admin Istrator
    </strong>
    <span
      className="note little-spacer-left"
    >
      admin
    </span>
  </td>
  <td
    className="text-right text-middle"
  >
    organization.members.x_groups.3
  </td>
  <td
    className="nowrap text-middle text-right"
  >
    <div
      className="dropdown"
    >
      <button
        className="dropdown-toggle little-spacer-right"
        data-toggle="dropdown"
      >
        <i
          className="icon-settings"
        />
         
        <i
          className="icon-dropdown"
        />
      </button>
      <ul
        className="dropdown-menu dropdown-menu-right"
      >
        <li>
          <ManageMemberGroupsForm
            member={
              Object {
                "avatar": "",
                "groupCount": 3,
                "login": "admin",
                "name": "Admin Istrator",
              }
            }
            organization={
              Object {
                "canAdmin": true,
                "key": "foo",
                "name": "Foo",
              }
            }
          />
        </li>
        <li
          className="divider"
          role="separator"
        />
        <li>
          <RemoveMemberForm
            member={
              Object {
                "avatar": "",
                "groupCount": 3,
                "login": "admin",
                "name": "Admin Istrator",
              }
            }
            organization={
              Object {
                "canAdmin": true,
                "key": "foo",
                "name": "Foo",
              }
            }
          />
        </li>
      </ul>
    </div>
  </td>
</tr>
`;
